mozjs-sys/etc/patches/0036-Redirect_errors_to_hilog_on_OpenHarmony.patch

```diff
@@ -10,6 +10,19 @@ index e0e14bf60..179de2592 100644
 
 
  @depends(target)
+@@ -1019,6 +1019,12 @@
+
+ set_config("ANDROID_CPU_ARCH", android_cpu_arch)
+
++@depends(target, when=target_is_ohos)
++def ohos_hilog_domain(target):
++    return "0xE0C4"
++
++set_define("OHOS_LOG_DOMAIN", ohos_hilog_domain)
```

Comment:
It looks like set_define [also accepts a `when` argument](https://github.com/servo/mozjs/blob/728acdf3d4ce0604e9f75dd1d539dc6f291ccec7/mozjs-sys/mozjs/build/moz.configure/arm.configure#L251). Is it not possible to use it here?

Reply:
I checked the implementation (set_define_impl) and the doc string says that when the value is false, the define will be explicitly undefined (-U).

Reply:
Ah, did you want to suggest to just remove the `ohos_hilog_domain` function and instead use when? that makes sense to me.

Reply:
Right, my idea was to see if we can just do `set_define("OHOS_LOG_DOMAIN", "0xE0C4", when=target_is_ohos)`